File: models/Employee.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const Employee = sequelize.define('Employee', {
    name: {
      type: DataTypes.STRING,
      allowNull: false
    },
    position: DataTypes.STRING,
    hourlyRate: DataTypes.FLOAT,
    employmentType: {
      type: DataTypes.ENUM('agency', 'dispatch', 'outsourcing'),
      allowNull: false
    },
    socialSecurityNumber: DataTypes.STRING,
    bankAccount: DataTypes.STRING
  });

  return Employee;
};

File: models/employee.go
package models

import "gorm.io/gorm"

type Employee struct {
	gorm.Model
	Name                 string  `json:"name" binding:"required"`
	Position             string  `json:"position" binding:"required"`
	HourlyRate           float64 `json:"hourlyRate" binding:"required"`
	ClientID             uint    `json:"clientId" binding:"required"`
	ContractID           uint    `json:"contractId" binding:"required"`
	EmploymentType       string  `json:"employmentType" binding:"required"`
	SocialSecurityNumber string  `json:"socialSecurityNumber" binding:"required"`
	BankAccount          string  `json:"bankAccount" binding:"required"`
}